### PR TITLE
fix: defaults options not correctly applied

### PR DESCRIPTION
**Pull request by SvMak**

### CODE REVIEW

**Review by Atinux — APPROVED** [review]

**Review by farzadso — APPROVED** [review]
LGTM!

**Comment by rxsto** [comment]
How are things looking here? 👀

**Comment by farzadso** [comment]
@rxsto The tests are failing. Please check it.

File: src/module.ts [hunk 1]
```diff
@@ -31,11 +31,7 @@ declare module '@nuxt/types' {
 export default <Module<Config>> function module (moduleOptions) {
   const { nuxt } = this
 
-  const options: Options = defu({
-    ...moduleOptions,
-    ...this.options.storage,
-    ...defaults
-  })
+  const options: Options = defu(moduleOptions, this.options.storage, defaults)
```
Comment (farzadso) [inline src/module.ts:34]:
```suggestion
  const options: DeepPartial<Options> = defu(moduleOptions, { ...this.options.storage }, defaults)
```

**Review by farzadso — CHANGES_REQUESTED** [review]
This will fix the tests. I tested it thoroughly and it seems like there were no issues.

File: src/module.ts (hunk 1 above)
Comment (SvMak) [reply src/module.ts]: Thanks a lot

**Pull request merged**